commcare_connect/connect_id_client/main.py

```diff
@@ -86,8 +87,14 @@ def get_user_otp(phone_number):
         response = _make_request(GET, "/users/generate_manual_otp", params={"phone_number": phone_number})
         data = response.json()
         return data.get("otp")
-    except httpx.HTTPError:
-        return None
+    except httpx.HTTPStatusError as e:
+        if e.response.status_code == 404:
```

Comment:
This doesn't really feel like a `ClientError`, but the expected result when a user types in an unknown number. I would recommend not raising an error in this part of the `if` block, just returning `None`, checking for that in the view and showing user not found in that case, then only raising for unexpected errors. That way, we know that we raise an exception when something went wrong in the code, not an expected status when the user makes a mistake.

Reply:
https://github.com/dimagi/commcare-connect/pull/710/commits/6a4d84baacd443f52e7bc1404c4c0b379abbce4d